release_tools/changelog.py

```diff
@@ -0,0 +1,329 @@
+    for a Git repository. You will need to run this script inside
+    that repository.
+
+    It will guide you to create a new entry. At the end of the process,
+    a text editor will open to let you review the entry and make the
+    final changes. You can skip some parts of the process providing
+    information in advance such as the title ('-t') or the category
+    ('-c') of the entry.
+
+    New entries will be stored in "releases/unreleased" directory.
+    This directory must be available under the Git root path. If you
+    don't want to create a new entry and see only the final result,
+    please active '--dry-run' flag.
+
+    In the case an entry with the same title already exists, an error
+    will be raised. Use '--overwrite' to force to replace the existing
+    entry.
+
+    You can also use this tool to create entries in a Git submodule.
+    Just run the script under the submodule directory.
+    """
+    click.echo()
+
+    dirpath = check_changelog_entries_dir()
+    content = create_changelog_entry_content(title, category,
+                                             run_editor=editor)
+
+    click.echo()
+
+    if dry_run:
+        click.echo(content)
+    else:
+        write_changelog_entry(dirpath, title, content,
+                              overwrite=overwrite)
+
+
+def check_changelog_entries_dir():
+    """Check and create the changelog directory."""
+
+    dirpath = determine_changelog_entries_dirpath()
+
+    if os.path.exists(dirpath):
+        return dirpath
+
+    click.echo("Error: Changelog entries directory does not exist.")
+
+    if not click.confirm("Do you want to create it?"):
+        msg = "Changelog entries directory is needed to continue."
+        raise click.ClickException(msg)
+
+    try:
+        os.makedirs(dirpath, mode=0o755)
+    except OSError as ex:
+        msg = "Unable to create directory. {}".format(str(ex))
+        raise click.ClickException(msg)
+    else:
+        click.echo("New directory {} created".format(dirpath))
+
+    return dirpath
+
+
+def create_changelog_entry_content(title, category, author=None, pr=None,
+                                   run_editor=True):
+    """Generates the content of a changelog entry."""
+
+    entry = ChangelogEntry(title, category, author=author, pr=pr)
+    contents = entry.to_dict()
+    stream = yaml.dump(contents, sort_keys=False,
+                       explicit_start=True)
+
+    # Allow the user to edit the final content of the entry
+    if run_editor:
+        content = click.edit(stream)
+    else:
+        content = stream
+
+    return content
+
+
+def write_changelog_entry(dirpath, title, content, overwrite=False):
+    """Store the contents of an entry in a file."""
+
+    filepath = determine_filepath(dirpath, title)
+
+    mode = 'w' if overwrite else 'x'
+
+    try:
+        filename = os.path.basename(filepath)
+
+        with open(filepath, mode=mode) as f:
+            f.write(content)
```

Comment:
An exception is thrown if the content is None. You can get this error if you modify the file and quit without saving.

Reply:
I've added some code to fix this situation. The script will raise an error when this happens. Unit tests updated accordingly.